### PR TITLE
chore: update Android project to Android Gradle Plugin 8.2

### PR DESCRIPTION
**Pull request by techouse**
## Summary
- updated project to use Android Gradle Plugin (AGP) **8.2**
- updated Kotlin to **1.9.22**
- added `android.namespace`
- removed `package` from AndroidManifest.xml
- bumped `compileSdkVersion` to **34**

 ## Testing Plan
 - [x] Tested locally.
 
Added this to pubspec.yaml
```yaml
mparticle_flutter_sdk:
  git:
    url: https://github.com/techouse/mparticle-flutter-sdk.git
    ref: chore/update-android-to-agp-8.2
```

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -25,18 +25,29 @@ apply plugin: 'com.android.library'
...
     }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_17
+    }
```
Comment (rmi22186) [inline android/build.gradle:41]: When I pulled this down and tried to run the example app in Android, I got the following error:

```
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':mparticle_flutter_sdk:compileDebugJavaWithJavac'.
> error: invalid source release: 17
```

Changing the 3 instances of `JavaVersion.VERSION_17` to `JavaVersion.VERSION_1_8`, or removing lines 34 through 41 resolved this for me and I was able to run the example app successfully.

**Review by rmi22186 — CHANGES_REQUESTED** [review]
Thank you for the PR.  some quick feedback below!

File: android/build.gradle (hunk 1 above)
Comment (techouse) [reply android/build.gradle]: Thanks for the feedback.

I have downgraded them all down to 1.8.

**Comment by techouse** [comment]
> Thank you for the PR. some quick feedback below!

Thanks for talking the time to review it @rmi22186 😇

I've amended the code as requested.

**Review by rmi22186 — DISMISSED** [review]

**Comment by techouse** [comment]
@rmi22186 looks like the build and test workflow is too outdated; I'll try to fix it but it will probably need another approval from you to run

**Comment by rmi22186** [comment]
@techouse yes i'll keep an eye on this PR!

**Comment by techouse** [comment]
> @techouse yes i'll keep an eye on this PR!

@rmi22186 ok, I have facelifted the project and fixed the CI

**Comment by techouse** [comment]
@rmi22186 I fixed the `Build example web` CI step.

**Review by rmi22186 — CHANGES_REQUESTED** [review]
Much appreciated for your input here.  Please see my fairly quick comments!

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -1,52 +1,155 @@
 name: Build and Test
-on: [push, pull_request]
+on:
```
[the remark this answers is not in the document]
Comment (techouse) [reply .github/workflows/pull-request.yml:2]: OK, I'll add it back to point to the latest Flutter.

File: example/.gitignore [hunk 3]
```diff
@@ -32,7 +32,6 @@
 /build/
 
 # Web related
-lib/generated_plugin_registrant.dart
```
[the remark this answers is not in the document]
Comment (techouse) [reply example/.gitignore:35]: Not sure how that got nuked. 🙈

[the remark this answers is not in the document]
Comment (techouse) [reply example/.gitignore]: Reverted.

File: .github/workflows/pull-request.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (techouse) [reply .github/workflows/pull-request.yml:2]: I added an env variable so that it's easier to change
```yaml
# pinned to latest stable version
FLUTTER_SDK_VERSION: 3.19.3
```

**Comment by techouse** [comment]
@rmi22186 

I added an env variable so that it's easier to track / change the Flutter version
```yaml
# pinned to latest stable version
FLUTTER_SDK_VERSION: 3.19.3
```

**Comment by techouse** [comment]
@rmi22186 I first decided to simply pin to the latest working version (current 3.19.3). Then I changed my mind to make a `matrix` of the latest stable version and last known working version, however, I later realized that it might not even compile when left unattended for a long time.

**tl;dr** I reverted back to a pinned version

```yaml
# pinned to latest stable version
FLUTTER_SDK_VERSION: 3.19.3
```

**Review by rmi22186 — APPROVED** [review]
Looks great.  Thank you for the contribution!

**Comment by rmi22186** [comment]
@techouse  - This has been published to pub.[dev](https://pub.dev/packages/mparticle_flutter_sdk) as version 1.0.4.  Thanks again!
